source/agora/node/Ledger.d

```diff
@@ -247,7 +254,13 @@ public class Ledger
 
     public void writeBlockFile (ref Block block) @safe
     {
-        auto fd = File("ledger.dat", "a"); // TODO move to config.yaml
+        auto dir = this.data_dir;
+        // Make sure we're in the right directory
+        if (!std.file.exists(dir) || !std.file.isDir(dir))
```

Comment:
And this should assert that it's a directory.

Reply:
add assert directory

Reply:
Now that you've added an assertion that it exists, this whole part should be changed into:

```
assert(dir.exists && std.file.isDir(dir));
```

There is no need to do `mkdir` if it already exists.

Reply:
Fixed.